src/sentry/snuba/sessions_v2.py

```diff
@@ -246,11 +246,16 @@ def __init__(self, query, project_ids=None):
             query_groupby.update(groupby.get_snuba_groupby())
         self.query_groupby = list(query_groupby)
 
-        params = {"project_id": project_ids or []}
+        # the `params` are:
+        # project_id, organization_id, environment;
+        # also: start, end; but we got those ourselves.
         snuba_filter = get_filter(self.query, params)
 
+        # this makes sure that literals in complex queries are properly quoted
+        conditions = [resolve_condition(c, lambda col: col) for c in snuba_filter.conditions]
```

Comment:
This column resolver isn't doing much. From the sentry error that came in, the root of the problem seems to be that the client didn't quote their query correctly as the release condition has a space in it.

The column was resolved to `message` because any unprefixed values map to the `message` field in our search grammar. Should there be checks applied here to ensure that all the conditions that are being used are for fields that exist? It looks like similar issues could be triggered if end users called the API directly with query strings like `invalid:value`.

Reply:
I think rejecting unknown fields is a good followup.
The issue being fixed here is that without that resolver, it would forward something like this to snuba: `["equals", ["release", "foo-bar"]]`, which snuba interprets as a column instead of literal. Passing stuff through that function transforms that to `["equals", ["release", "\"foo-bar\""]]` which snuba correctly treats as literal.

TBH, I still don’t know enough about how to use that stuff *correctly*.

Reply:
Won't you still be getting an invalid field in the search though? The original problem was caused by `release:1234 (456)` which is interpreted as `[[equals, [release, 1234]], [equals, [message, 456]]]` I think quoting the release field will help part of the problem, but without quoting values in the client query you'll still be hitting invalid queries as the message field doesn't exist.

Reply:
Ah, now I understand. That is a separate issue I guess…